src/solver/type_getters.jl

```diff
@@ -183,18 +164,27 @@ function get_spaces(parsed_args, params, comms_ctx)
         else
             Meshes.Uniform()
         end
-        if warp_function == nothing
+        if warp_function == nothing && topography != "Earth"
             make_hybrid_spaces(h_space, z_max, z_elem, z_stretch; deep)
-        else
+        elseif warp_function != nothing && topography != "Earth"
             make_hybrid_spaces(
                 h_space,
                 z_max,
                 z_elem,
                 z_stretch;
-                parsed_args = parsed_args,
                 surface_warp = warp_function,
                 deep,
             )
+        elseif topography == "Earth"
```

Comment:
We can first check topography here, then check warp_function, and can get rid of the `&& topography != "Earth"`? I think the logic is more clear this way.

Reply:
Yes ; I'll go over this again and try to clean up some of these branching statements in `common_spaces` and `type_getters.`